app/services/form1010cg/service.rb

```diff
@@ -108,7 +109,10 @@ def submit_attachment # rubocop:disable Metrics/MethodLength
     #
     # @return [nil]
     def assert_veteran_status
-      raise InvalidVeteranStatus if icn_for('veteran') == NOT_FOUND
+      if icn_for('veteran') == NOT_FOUND
+        log_invalid_veteran_status
```

Comment:
Rather than being in the `Form1010cg::Service`, I think all if this logic could live in `Form1010cg::Auditor`. It already has a method there to log and increment StatsD for the same event: when the veterans status is not valid.

Something like:
```rb
# app/services/form1010cg/auditor.rb
# ...
  # ...
    def record_submission_failure_client_qualification(claim_guid:, veteran_name:)
      increment self.class.metrics.submission.failure.client.qualification
      log 'Submission Failed: qualifications not met', claim_guid: claim_guid, veteran_name: veteran_name
      notify_ga_invalid_veteran_status
    end
    # ...
    private

    def get_subdomain
      # ...
    end

    def notify_ga_invalid_veteran_status
      # ...
    end
  # ...
# ...
```

`#record_submission_failure_client_qualification` can just accept the `ga_client_id` as an extra argument without the need to attach it to the claim.

For context, this record method is called here: https://github.com/department-of-veterans-affairs/vets-api/blob/59dbf80673ac8802ada3f8768538426420bd6fa4/app/controllers/v0/caregivers_assistance_claims_controller.rb#L8

Reply:
> Lastly, what happens if the x-google-client-id header is not present?

how about i don't log the event if the client id is not present?

Reply:
Yes, that works 👍